examples/adt-tests.dx

```diff
@@ -209,3 +209,59 @@ def catLists (xs:List a) (ys:List a) : List a =
   n = 1 + 4
   AsList _ (for i:(Fin n). ordinal i)
 > (AsList 5 [0, 1, 2, 3, 4])
+
+
+
+def listToTable ((AsList n xs): List a) : (Fin n)=>a = xs
+
+:t listToTable
+> ((a:Type) ?-> (pat:(List a)) -> (Fin (%projectElt [0] pat)) => a)
+
+:p
+  l = AsList _ [1, 2, 3]
+  sum $ listToTable l
+> 6
+
+def listLength ((AsList length xs):List a) : Int = length
+def listToTable2 (l: List a) : (Fin (listLength l))=>a =
+  (AsList _ xs) = l
+  xs
+
+:t listToTable2
+> ((a:Type) ?-> (l:(List a)) -> (Fin (%projectElt [0] l)) => a)
+
+:p
+  l = AsList _ [1, 2, 3]
+  sum $ listToTable2 l
+> 6
+
+data Graph a:Type =
+  MkGraph n:Type nodes:(n=>a) m:Type edges:(m=>(n & n))
+
+def graphToAdjacencyMatrix ((MkGraph n nodes m edges):Graph a) : n=>n=>Bool =
+  init = for i j. False
+  snd $ withState init \mRef.
+    for i:m.
+      (from, to) = edges.i
+      mRef!from!to := True
+
+:t graphToAdjacencyMatrix
+> ((a:Type)
+>  ?-> (pat:(Graph a)) -> (%projectElt [0] pat) => (%projectElt [0] pat) => Bool)
+
+:p
+  g : Graph Int = MkGraph (Fin 3) [5, 6, 7] (Fin 4) [(0@_, 1@_), (0@_, 2@_), (2@_, 0@_), (1@_, 1@_)]
+  graphToAdjacencyMatrix g
+> [[False, True, True], [False, True, False], [True, False, False]]
+
+
+def deepUnpack (x:MyPair a (MyPair (MyIntish & b) c)) : Int =
+  (MkMyPair _ (MkMyPair (MkIntish y, _) _)) = x
+  y
```

Comment:
It ends up producing `ProjectElt` atoms with long lists of ints, whereas most tests just have one or two levels of projection. So it's intended to make sure we are treating the integer lists in a consistent way.

Reply:
How does it produce the `ProjectElt`? Because those are the atoms that variables bound in patterns get assigned to?

Can you please add a comment explaining this in a little more detail. In a few months none of us will remember why this test case is even here.